lib/ElasticSearch/Event/EventManager.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ *
+ * @author      Michal Maszkiewicz
+ * @package     Elastic Search Plugin
+ */
+
+namespace ElasticSearch\Event;
+
+use Closure;
+use Document_Page;
+use ElasticSearch\Job\CacheAllPagesJob;
+use ElasticSearch\Repository\PageRepository;
+use Schedule_Maintenance_Job;
+use Schedule_Manager_Procedural;
+use Zend_EventManager_Event as Event;
+use Zend_EventManager_EventManager;
+use Zend_EventManager_Exception_InvalidArgumentException;
+
+
+class EventManager
+{
+    /**
+     * 
+     */
+    const MAINTENANCE_JOB_REBUILD_PAGES = 'elasticsearch-recache-pages';
+    
+    /**
+     * @var Zend_EventManager_EventManager
+     */
+    protected $pimcoreEventManager;
+
+    /**
+     * @var PageRepository
+     */
+    protected $pageRepository;
+    
+    /**
+     *
+     * @var CacheAllPagesJob
+     */
+    protected $cacheAllPagesJob;
+    
+    
+    
+    /**
+     * @param Zend_EventManager_EventManager $pimcoreEventManager
+     * @param PageRepository $pageRepository
+     */
+    public function __construct(
+        Zend_EventManager_EventManager $pimcoreEventManager,
+        PageRepository $pageRepository,
+        CacheAllPagesJob $cacheAllPagesJob
+    ) {
+        $this->pimcoreEventManager = $pimcoreEventManager;
+        $this->pageRepository = $pageRepository;
+        $this->cacheAllPagesJob = $cacheAllPagesJob;
+    }
+
+    /**
+     * Attached a maintenance 
+     */
+    public function attachMaintenance()
+    {
+        $this->pimcoreEventManager->attach(
+            'system.maintenance',
+            Closure::bind(
+                function(Event $event) {
+                    /* @var $target Schedule_Manager_Procedural */
+                    $target = $event->getTarget();
+
+                    $target->registerJob(
+                        new Schedule_Maintenance_Job(
+                            self::MAINTENANCE_JOB_REBUILD_PAGES,
+                            $this->cacheAllPagesJob,
+                            'rebuildPageCache'
+                        )
+                    );
+                },
+                $this
```

Comment:
Sadly not; I have tried this before.  The only other option I'm aware of is to initialise a variable which isn't a reserved keyword like $this (say, $self) and assign it the $this reference, then "use" $self.  Sucks!

Whereas you could just bind the closure; really useful PHP5.4+ feature.